drivers/net/bnxt/bnxt_ethdev.c

```diff
@@ -1140,13 +1176,17 @@ static int bnxt_set_vf_vlan_filter_op(struct rte_eth_dev *dev, uint16_t vlan,
 				}
 				if (j == bp->pf.vf_info[i].vlan_count) {
 					/* Now check that there's space */
-					if (bp->pf.vf_info[i].vlan_count == getpagesize() / sizeof(struct bnxt_vlan_table_entry)) {
-						RTE_LOG(ERR, PMD, "VF %d VLAN table is full, cannot add VLAN %u\n", i, vlan);
```

Comment:
Basically as you said, the structures can't change, and this one will always run out first, so the other check will always be false.  No point in testing something we know.